build-tests/api-extractor-test-05/etc/api-extractor-test-05.api.json

```diff
@@ -0,0 +1,289 @@
+        "@privateRemarks": true,
+        "@public": true,
+        "@readonly": true,
+        "@remarks": true,
+        "@returns": true,
+        "@sealed": true,
+        "@see": true,
+        "@throws": true,
+        "@typeParam": true,
+        "@virtual": true,
+        "@betaDocumentation": true,
+        "@internalRemarks": true,
+        "@preapproved": true
+      }
+    }
+  },
+  "kind": "Package",
+  "canonicalReference": "api-extractor-test-05!",
+  "docComment": "",
+  "name": "api-extractor-test-05",
+  "members": [
+    {
+      "kind": "EntryPoint",
+      "canonicalReference": "api-extractor-test-05!",
+      "name": "",
+      "members": [
+        {
+          "kind": "Namespace",
+          "canonicalReference": "api-extractor-test-05!testJsLibrary:namespace",
+          "docComment": "/**\n * @public\n */\n",
+          "excerptTokens": [
+            {
+              "kind": "Content",
+              "text": "export namespace testJsLibrary "
+            }
+          ],
+          "releaseTag": "Public",
+          "name": "testJsLibrary",
+          "members": [
+            {
+              "kind": "Variable",
+              "canonicalReference": "api-extractor-test-05!testJsLibrary.MY_CONSTANT:var",
+              "docComment": "/**\n * This is my constant.\n *\n * @deprecated\n *\n * This is deprecated for a good reason.\n */\n",
+              "excerptTokens": [
+                {
+                  "kind": "Content",
+                  "text": "MY_CONSTANT : "
+                },
+                {
+                  "kind": "Content",
+                  "text": "number"
+                }
+              ],
+              "releaseTag": "Public",
+              "name": "MY_CONSTANT",
+              "variableTypeTokenRange": {
+                "startIndex": 1,
+                "endIndex": 2
+              }
+            },
+            {
+              "kind": "Class",
+              "canonicalReference": "api-extractor-test-05!testJsLibrary.MyClass:class",
+              "docComment": "/**\n * This is my class.\n */\n",
+              "excerptTokens": [
+                {
+                  "kind": "Content",
+                  "text": "export class MyClass "
+                }
+              ],
+              "releaseTag": "Public",
+              "name": "MyClass",
+              "members": [
+                {
+                  "kind": "Method",
+                  "canonicalReference": "api-extractor-test-05!testJsLibrary.MyClass#someMethod:member(1)",
+                  "docComment": "/**\n * This method does something.\n *\n * @param x - This is x.\n *\n * @param y - This is y.\n */\n",
+                  "excerptTokens": [
+                    {
+                      "kind": "Content",
+                      "text": "someMethod(x : "
+                    },
+                    {
+                      "kind": "Content",
+                      "text": "number "
+                    },
```

Comment:
I think in an ideal world, we'd want this:

```
{
    "kind": "Content",
    "text": "someMethod(x : "
},
{
    "kind": "Content",
    "text": "number "
},
```

to really be this:

```
{
    "kind": "Content",
    "text": "someMethod(x : "
},
{
    "kind": "Content",
    "text": "number"
},
{
    "kind: "Content",
    "text": " "
}
```

That way it's possible to construct a param type range that only grabs `number` and not `number `. However, my guess is that this may be non-trivial work and potentially also not really worth it at this time if the tool producing the `.d.ts` is already adding _extraneous_ spaces. So, feel free to ack & close, but just thought I'd mention this.

Reply:
I'm not sure offhand if there'd be a clean way to handle that, unless we want to only lead to teh additional spaces for the last of a token range to add the space to the next, since I could see the argument for `someMethod(x : ` to have teh last space moved to the next token.

Reply:
I'm most concerned that the normalized compiler output is handled correctly, since this is our primary use case. If people want to do additional work to normalize the output, that's cool.

HOWEVER there was some discussion about running a preprocessor such as Prettier on the .d.ts content (before and/or after) API Extractor processes them.  That could be a "battle axe" that gets us out of having to deal with a lot of these edge cases.  (Code formatting is a lot trickier than it seems!)